scenarios/aws/eks/run.go

```diff
@@ -132,29 +135,59 @@ func Run(ctx *pulumi.Context) error {
 		comp.KubeConfig = cluster.KubeconfigJson
 
 		nodeGroups := make([]pulumi.Resource, 0)
+		ngMutex := sync.Mutex{}
+		cgroupV1Ng := pulumi.StringArray{}
+		cgroupV2Ng := pulumi.StringArray{}
+		wg := sync.WaitGroup{}
 		// Create managed node groups
 		if awsEnv.EKSLinuxNodeGroup() {
+			wg.Add(1)
 			ng, err := localEks.NewLinuxNodeGroup(awsEnv, cluster, linuxNodeRole)
 			if err != nil {
 				return err
 			}
 			nodeGroups = append(nodeGroups, ng)
+			// The default AMI used for Amazon Linux 2 is using cgroupv1
+			ng.NodeGroup.NodeGroupName().ApplyT(func(s pulumi.String) interface{} {
+				ngMutex.Lock()
+				defer ngMutex.Unlock()
+				cgroupV1Ng = append(cgroupV1Ng, s)
+				wg.Done()
+				return nil
+			})
 		}
```

Comment:
Pulumi must be able to execute the full `Run()` function before starting the resources instantiation. We shouldn’t delay some resources declaration until some others are instantiated.
We should directly pass the `ng.NodeGroup.NodeGroupName()` to the `tracegen.K8sAppDefinition` function.

Reply:
Fixed in the last commit